sdk/search/search-documents/src/searchServiceClient.ts

```diff
@@ -153,7 +153,9 @@ export class SearchServiceClient {
    * Retrieves a list of existing indexes in the service.
    * @param options Options to the list index operation.
    */
-  public async listIndexes(options: ListIndexesOptions = {}): Promise<Index[]> {
+  public async listIndexes<Fields extends keyof Index>(
+    options: ListIndexesOptions<Fields> = {}
+  ): Promise<Index[]> {
```

Comment:
I think the return type has to be `Promise<Pick<Array<Index,Fields>>>` for it to actually remove the non-selected items, same for the other results